proto/cacheclient.proto

```diff
@@ -435,6 +437,47 @@ message _ListLengthResponse {
   }
 }
 
+message _ListTrimRequest {
+  bytes list_name = 1;
+  optional sint32 start = 2;
+  optional sint32 end = 3;
+  optional sint32 step = 4;
+}
+
+message _ListTrimResponse {
+  message _Found {
+    repeated bytes values = 1;
+  }
+
+  message _Missing {}
+
+  oneof list {
+    _Found found = 1;
+    _Missing missing = 2;
+  }
+}
+
+message _ListRangeRequest {
+  bytes list_name = 1;
+  optional sint32 start = 2;
+  optional sint32 end = 3;
+  optional sint32 step = 4;
```

Comment:
I think in the barest definition of this API we need simply return the range between start and end - what is step intended to do here? Do we foresee a use case for this?

Reply:
I agree with this, was going to ask the same.